packages/ramldoc-generator/src/main.js

```diff
@@ -0,0 +1,40 @@
+const fs = require('fs');
+const path = require('path');
+const shell = require('shelljs');
+const chalk = require('chalk');
+
+function generateRaml(options) {
+  const resolvedSourcePath = path.resolve(options.apiSpecSourcePath);
+
+  // Check if the spec file exists.
+  if (!fs.existsSync(resolvedSourcePath)) {
+    console.error(
+      `"${resolvedSourcePath}" does not exist`,
+      chalk.red.bold('ERROR')
+    );
+    process.exit(1);
+  }
+
+  generateRamlIfSpecPath({
+    ...options,
+    apiSpecSourcePath: resolvedSourcePath,
+  });
+}
+
+function generateRamlIfSpecPath(options) {
+  const resolvedDestinationPath = path.resolve(options.apiSpecDestinationPath);
+
+  console.log(
+    `RAML docs will be generated in this directory: ${chalk.blue(
+      resolvedDestinationPath
+    )}`
+  );
+
+  const jarFile = path.resolve(__dirname, '../jar/rmf-codegen.jar');
```

Comment:
```suggestion
  const jarFile = path.join(__dirname, '../jar/rmf-codegen.jar');
```

Reply:
Why is this better?

Reply:
Why is this better?

Reply:
`__dirname` is already an absolute path, and you just want to append other partial paths.

That's what `path.join` is for. `path.resolve` does not work as you would expect in this case. See below:

```
> path.resolve(process.cwd(), '/foo')
'/foo'
> path.join(process.cwd(), '/foo')
'/Users/emmenko/dev/src/foo'

> path.resolve(process.cwd(), '..', '/foo')
'/foo'
> path.join(process.cwd(), '..', '/foo')
'/Users/emmenko/dev/foo'
```

Reply:
Why use `process.cwd()`?

Anyway, in my case it works exactly as expected.

Reply:
process.cwd() was just for the example, as I run the code in the node repl and __dirname is not available there.

Using path.resolve is fine if you know what you’re doing, but it might lead to unexpected results.
That’s why usually for building paths you would use path.join.